Dockerfile.centos

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2012-2016 Red Hat, Inc
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#   Dharmit Shah  - Initial implementation
+#   Mario Loriedo - Improvements
+#
+# To build it, run in the repository root:
+#  `docker build -t registry.centos.org/eclipse/che-server -f Dockerfile.centos .`
+#
+# To run it:
+#  docker run --net=host \
+#             --name che \
+#             -v /var/run/docker.sock:/var/run/docker.sock \
+#             -v /home/user/che/lib:/home/user/che/lib-copy \
+#             -v /home/user/che/workspaces:/home/user/che/workspaces \
+#             -v /home/user/che/storage:/home/user/che/storage \
+#             registry.centos.org/eclipse/che-server
+#           
+FROM registry.centos.org/centos/centos:latest
+
+ENV LANG=C.UTF-8 \
+    JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.102-1.b14.el7_2.x86_64/jre \
+    PATH=${PATH}:${JAVA_HOME}/bin \
+    CHE_HOME=/home/user/che \
+    DOCKER_VERSION=1.6.0 \
+    DOCKER_BUCKET=get.docker.com
+
+RUN yum -y update update && \
+    yum -y install ca-certificates curl openssl java sudo && \
+    curl -sSL "https://${DOCKER_BUCKET}/builds/Linux/x86_64/docker-${DOCKER_VERSION}" -o /usr/bin/docker && \
+    chmod +x /usr/bin/docker
+RUN useradd user -d /home/user -s /bin/bash -u 1000 -G root && \
+    groupadd docker -g 101 && \
+    usermod -aG ftp,users,docker user && \
+    echo "%root ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
+    sed -i 's/Defaults    requiretty/#Defaults    requiretty/g' /etc/sudoers && \
+    rm -rf /tmp/* /var/cache/yum
+
+EXPOSE 8000 8080
+
+USER user
+
+ENV CHE_ARTIFACT_URL="https://install.codenvycorp.com/che/eclipse-che-nightly.tar.gz"
+
+RUN curl -sSL "$CHE_ARTIFACT_URL" -o /home/user/eclipse-che.tar.gz
+RUN tar -xzf /home/user/eclipse-che.tar.gz -C /home/user/
+RUN rm /home/user/eclipse-che.tar.gz
+RUN mv /home/user/eclipse-che-* /home/user/che
```

Comment:
to avoid several layers, RUN instructions should be merged into one (and tar command should be for example piped to curl (no need to remove at the end, etc)